File: .ProgramingBasics/Simple-Loops/10.HalfSumElement/src/HalfSumElement.java
import java.util.Scanner;

public class HalfSumElement {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        int sum = 0;
        int max = 0;

        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(scanner.nextLine());

            if (num > max) {
                max = num;
            }

            sum += num;
        }

        sum -= max;

        if (sum == max) {
            System.out.println("Yes");
            System.out.println("Sum = " + sum);
        }
        else {
            int diff = Math.abs(max - sum);
            System.out.println("No");
            System.out.println("Diff = " + diff);
        }
    }
}

File: ProgramingBasics/SimpleLoops/10.HalfSumElement/10.HalfSumElement/HalfSumElement.cs
using System;

namespace _10.HalfSumElement
{
    public class HalfSumElement
    {
        public static void Main(string[] args)
        {
            var n = int.Parse(Console.ReadLine());
            var sum = 0;
            var max = 0;

            for (var i = 0; i < n; i++)
            {
                var num = int.Parse(Console.ReadLine());

                if (num > max)
                {
                    max = num;
                }

                sum += num;
            }

            sum -= max;

            if (sum == max)
            {
                Console.WriteLine("Yes");
                Console.WriteLine($"Sum = {sum}");
            }
            else
            {
                var diff = Math.Abs(max - sum);
                Console.WriteLine("No");
                Console.WriteLine($"Diff = {diff}");
            }
        }
    }
}
